src/domain/dataset-update-flow/src/aggregates/update.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use event_sourcing::*;
+use kamu_task_system::TaskID;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Aggregate, Debug)]
+pub struct Update(Aggregate<UpdateState, (dyn UpdateEventStore + 'static)>);
+
+impl Update {
+    /// Creates a dataset update process
+    pub fn new(update_id: UpdateID, dataset_id: DatasetID, trigger: UpdateTrigger) -> Self {
+        Self(
+            Aggregate::new(
+                dataset_id.clone(),
+                UpdateInitiated {
+                    event_time: Utc::now(),
+                    update_id,
+                    dataset_id,
+                    trigger,
+                },
+            )
+            .unwrap(),
+        )
+    }
+
+    /// Postpone launching for now
+    pub fn postpone(
```

Comment:
I'm having trouble imagining how this API function will be used.

In my mind there will be a service controlling dataset updates:
* It listens on event bus for evens like `DatasetUpdated`, `ScheduledCommand<DatasetUpdatePolling>`, etc.
* It translates thes events into `Updatetrigger`s
* Checks if there is already an `Updated` scheduled for a dataset or creates a new one
* It then makes a decision on when to run the update (based on all triggers) and updates the state of an `Update` aggregate

So I imagine something like `fn add_trigger(trigger)` and `fn set_start_condition(condition)`.

Reply:
Accepted the suggested naming